docs/modules/ROOT/pages/explanations/commodore-components/kustomizations.adoc

```diff
@@ -0,0 +1,126 @@
+    manifests_version: '1.0.0'
+    images:
+      fancy-sli-controller:
+        registry: quay.io
+        repository: syn/fancy-sli-controller
+        tag: '1.0.0'
+    kustomize_input:
+      namePrefix: syn-slo-
+      namespace: ${slo:namespace}
+----
+<1> If the kustomization is stored in a subpath of the repository, use `//` to separate the repository URL from the path inside the repository.
+
+In `component/fancy-sli-controller-overlay.jsonnet`, we render a `kustomization.yaml` which refers to the kustomization specified in `class/defaults.yml` and ensures the resulting Kubernetes resources use the image registry, repository and tags as specified in the parameters.
+We also merge the provided `kustomize_input` into the `kustomization.yaml`.
+This allows users to make arbitrary changes to the kustomization.
+
+.component/fancy-sli-controller-overlay.jsonnet
+[source,jsonnet]
+----
+// Template to render kustomization overlay
+local com = import 'lib/commodore.libjsonnet';
+local kap = import 'lib/kapitan.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.slo;
+
+local image = params.images.fancy_sli_controller;
+
+com.Kustomization( <1>
+  params.kustomization_url,
+  params.manifests_version,
+  {
+    'quay.io/syn/fancy-sli-controller': {
+      newTag: image.tag,
+      newName: '%(registry)s/%(repository)s' % image,
+    },
+  },
+  params.kustomize_input,
+)
+----
+<1> This function generates an object which is suitable as a Jsonnet output.
+The resulting output will consist of a single `kustomization.yaml` file which is suitable as a `kustomize build` input.
+The provided `kustomization_url` and `manifests_version` will be rendered as an entry in the kustomization's `resource` field.
+The contents of the third parameter will be used to render entries for the kustomization's `images` field.
+See the xref:commodore:ROOT:reference/commodore-libjsonnet.adoc#_kustomizationbase_url_base_version_images_kustomize_input[`commodore.libjsonnet`] documentation for details.
+
+In `class/slo.yml`, we render the `kustomization.yaml` and pass it `kustomize build` to generate the Kubernetes resources to commit to the cluster catalog.
+
+.class/slo.yml
+[source,yaml]
+----
+parameters:
+  kapitan:
+    inputs:
+      # render kustomization overlay into <component dir>/fancy-sli-controller
+      - input_paths:
+          - ${_base_directory}/component/fancy-sli-controller-overlay.jsonnet
+        input_type: jsonnet
+        output_path: ${_base_directory}/fancy-sli-controller
+
+      # Run kustomize on the rendered overlay.
+      # NOTE: Kapitan ignores `output_path` for `input_type=external`
+      - input_type: external
+        output_path: .
+        input_paths:
+          - ${_kustomize_wrapper} <1>
+        env_vars:
+          INPUT_DIR: ${_base_directory}/fancy-sli-controller <2>
+        args:
+          - \${compiled_target_dir}/${_instance}/fancy-sli-controller-deployment <3>
```

Comment:
Because then `${compiled_target_dir}` doesn't get replaced by Kapitan, cf. https://github.com/kapicorp/kapitan/blob/257f53962d375be12da74d9c966e6e423aa68a90/kapitan/inputs/external.py#L48

Reply:
🤷‍♀️